Source/WebCore/accessibility/AXObjectCache.h

```diff
@@ -139,6 +139,110 @@ class AccessibilityReplacedText {
     VisiblePositionIndexRange m_replacedRange;
 };
 
+#define WEBCORE_AXNOTIFICATION_KEYS_DEFAULT(macro) \
+    macro(AccessKeyChanged) \
```

Comment:
This looks strictly worse. I don't understand why we want to adopt this change.

Reply:
I linked the explanation above but I can link it again here: https://github.com/WebKit/WebKit/pull/23035#discussion_r1494036159

Introducing `WebKit/UIProcess/API/C/WKAXTypes.h` with enums like...
```cpp
enum AXNotification {
     // must be same as WebCore::AXObjectCache::AXNotification
     kWKAXAccessKeyChanged,
     ...
 };
 typedef uint32_t WKAXNotification;
 ```
 ...and extending `WebKit/UIProcess/API/C/WKAPICast.h` with casts like...
 ```cpp
 inline WKAXNotification toAPI(WebCore::AXObjectCache::AXNotification notification)
 {
     switch (notification) {
     case WebCore::AXObjectCache::AXNotification::AXAccessKeyChanged:
         return kWKAXAccessKeyChanged;
     ...
     }
 }
```
...feels rather hard to justify if only one platform is planning to use these casts.

Since the idea is just to ensure that the API casts get updated whenever the WebCore enums get updated, the ideal situation would be to have this happen automatically, without requiring developers to be conscious of it. This automation is achieved by [X macros](https://en.wikipedia.org/wiki/X_macro), which allow us to do an exhaustive `switch` on a enum without being concerned with the specific values of that enum changing. Indeed, forgetting about any platform-specific needs, in AXLogger, we're able to reduce a `switch` from 286 lines to just 8.

I understand that while this is commonplace in JSC (from a quick search it seems like we have almost 200 of them), it surely feels quite foreign in WebCore. But I'm not sure of a different tool that fits this problem so well.

Reply:
Seems like I've overestimated how foreign this actually is outside of JSC. We have X macros defined in...
- [`Source/WebCore/bindings/js/WebCoreBuiltinNames.h`](https://github.com/WebKit/WebKit/blob/main/Source/WebCore/bindings/js/WebCoreBuiltinNames.h#L38-L740)
- [`Source/WebCore/cssjit/SelectorCompiler.cpp`](https://github.com/WebKit/WebKit/blob/main/Source/WebCore/cssjit/SelectorCompiler.cpp#L85-L150)
- [`Source/WebCore/html/parser/HTMLDocumentParserFastPath.cpp`](https://github.com/WebKit/WebKit/blob/main/Source/WebCore/html/parser/HTMLDocumentParserFastPath.cpp#L137-L155)
- [`Source/WebCore/inspector/InspectorCanvasCallTracer.h`](https://github.com/WebKit/WebKit/blob/main/Source/WebCore/inspector/InspectorCanvasCallTracer.h#L84-L190)
- [`Source/WebCore/inspector/InstrumentingAgents.h`](https://github.com/WebKit/WebKit/blob/main/Source/WebCore/inspector/InstrumentingAgents.h#L110-L138)
- [`Source/WebCore/page/ResourceUsageData.h`](https://github.com/WebKit/WebKit/blob/main/Source/WebCore/page/ResourceUsageData.h#L37-L58)
- [`Source/WebCore/platform/CommonAtomStrings.h`](https://github.com/WebKit/WebKit/blob/main/Source/WebCore/platform/CommonAtomStrings.h#L33-L77)
- [`Source/WebCore/platform/Logging.h`](https://github.com/WebKit/WebKit/blob/main/Source/WebCore/platform/Logging.h#L39-L151)
- [`Source/WebCore/platform/graphics/avfoundation/MediaPlayerPrivateAVFoundation.h`](https://github.com/WebKit/WebKit/blob/main/Source/WebCore/platform/graphics/avfoundation/MediaPlayerPrivateAVFoundation.h#L81-L107)
- [`Source/WebCore/platform/graphics/texmap/TextureMapperShaderProgram.h`](https://github.com/WebKit/WebKit/blob/main/Source/WebCore/platform/graphics/texmap/TextureMapperShaderProgram.h#L35-L74)
- [`Source/WebCore/PAL/pal/Logging.h`](https://github.com/WebKit/WebKit/blob/main/Source/WebCore/PAL/pal/Logging.h#L40-L48)
- [`Source/WebKit/Platform/Logging.h`](https://github.com/WebKit/WebKit/blob/main/Source/WebKit/Platform/Logging.h#L41-L133)
- [`Source/WebKit/UIProcess/ios/WKContentViewInteraction.h`](https://github.com/WebKit/WebKit/blob/main/Source/WebKit/UIProcess/ios/WKContentViewInteraction.h#L157-L236)
- [`Source/WTF/wtf/Logging.h`](https://github.com/WebKit/WebKit/blob/main/Source/WTF/wtf/Logging.h#L39-L52)
- [`Source/WTF/wtf/SystemTracing.h`](https://github.com/WebKit/WebKit/blob/main/Source/WTF/wtf/SystemTracing.h#L219-L227)
- [`Tools/TestRunnerShared/EventSerialization/mac/EventSerializerMac.mm`](https://github.com/WebKit/WebKit/blob/main/Tools/TestRunnerShared/EventSerialization/mac/EventSerializerMac.mm#L87-L152)
- [`Tools/TestWebKitAPI/Tests/WebCore/DisplayListRecorderTests.cpp`](https://github.com/WebKit/WebKit/blob/main/Tools/TestWebKitAPI/Tests/WebCore/DisplayListRecorderTests.cpp#L107-L125)

...and a few more files which are GTK/WPE-specific.

So I revoke my hesitation; this is not only the right tool for the job, but it's also a perfectly normal thing for us to do.